src/task/executable_task.rs

```diff
@@ -328,3 +329,155 @@ fn get_output_writer_and_handle() -> (ShellPipeWriter, JoinHandle<String>) {
     let handle = reader.pipe_to_string_handle();
     (writer, handle)
 }
+
+/// Task specific environment variables.
+fn get_export_specific_task_env(task: &Task) -> String {
+    // Append the environment variables if they don't exist
+    let mut export = String::new();
+    if let Some(env) = task.env() {
+        for (key, value) in env {
+            if value.contains(format!("${}", key).as_str()) || std::env::var(key.as_str()).is_err()
+            {
+                tracing::info!("Setting environment variable: {}=\"{}\"", key, value);
+                export.push_str(&format!("export \"{}={}\";\n", key, value));
+            } else {
+                tracing::info!("Environment variable {} already set", key);
+            }
+        }
+    }
+    export
+}
+
+/// Determine the environment variables to use when executing a command. The method combines the
+/// activation environment with the system environment variables.
+pub async fn get_task_env<'p>(
+    environment: &Environment<'p>,
+    clean_env: bool,
+) -> miette::Result<HashMap<String, String>> {
+    // Make sure the system requirements are met
+    verify_current_platform_has_required_virtual_packages(environment).into_diagnostic()?;
+
+    // Get environment variables from the activation
+    let env_var_behavior = if clean_env {
+        CurrentEnvVarBehavior::Clean
+    } else {
+        CurrentEnvVarBehavior::Include
+    };
+    let mut activation_env = await_in_progress("activating environment", |_| {
+        environment
+            .project()
+            .get_activated_environment_variables(environment, env_var_behavior)
+    })
+    .await
+    .wrap_err("failed to activate environment")?
+    .clone();
+
+    // Add the current working directory to the environment
+    activation_env.insert(
+        "INIT_CWD".to_string(),
+        std::env::current_dir()
+            .into_diagnostic()
+            .wrap_err("failed to get current working directory")?
+            .to_string_lossy()
+            .to_string(),
+    );
```

Comment:
How about changing this to only add the `INIT_CWD` variable if the `current_dir` is available instead of erroring?

Reply:
Done